tests/test-commit.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+#
+# Copyright (C) 2016 William Manley <will@williammanley.net>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+
+set -euo pipefail
+
+. $(dirname $0)/libtest.sh
+
+echo "1..1"
+
+setup_test_repository "bare"
+
+$OSTREE commit --tree=dir=$test_srcdir/tree-A -b A -m A
+$OSTREE commit --tree=dir=$test_srcdir/tree-B -b B -m B
+
```

Comment:
does the commit miss some files?

Anyway, it would probably be cleaner to just do in the test itself:

mkdir tree-A tree-B
touch tree-A/file-a tree-B/file-b